com.unity.multiplayer.mlapi/Runtime/Core/NetworkManager.cs

```diff
@@ -1600,51 +1600,7 @@ internal void HandleApproval(ulong ownerClientId, bool createPlayerObject, uint?
 
                         for (int i = 0; i < m_ObservedObjects.Count; i++)
                         {
-                            var observedObject = m_ObservedObjects[i];
-                            writer.WriteBool(observedObject.IsPlayerObject);
-                            writer.WriteUInt64Packed(observedObject.NetworkObjectId);
-                            writer.WriteUInt64Packed(observedObject.OwnerClientId);
-
-                            NetworkObject parent = null;
-
-                            if (!observedObject.AlwaysReplicateAsRoot && observedObject.transform.parent != null)
-                            {
-                                parent = observedObject.transform.parent.GetComponent<NetworkObject>();
-                            }
-
-                            if (parent == null)
-                            {
-                                writer.WriteBool(false);
-                            }
-                            else
-                            {
-                                writer.WriteBool(true);
-                                writer.WriteUInt64Packed(parent.NetworkObjectId);
-                            }
-
-                            writer.WriteBool(observedObject.IsSceneObject ?? true);
-                            writer.WriteUInt32Packed(observedObject.GlobalObjectIdHash);
-
-                            if (observedObject.IncludeTransformWhenSpawning == null || observedObject.IncludeTransformWhenSpawning(ownerClientId))
-                            {
-                                writer.WriteBool(true);
-                                writer.WriteSinglePacked(observedObject.transform.position.x);
-                                writer.WriteSinglePacked(observedObject.transform.position.y);
-                                writer.WriteSinglePacked(observedObject.transform.position.z);
-
-                                writer.WriteSinglePacked(observedObject.transform.rotation.eulerAngles.x);
-                                writer.WriteSinglePacked(observedObject.transform.rotation.eulerAngles.y);
-                                writer.WriteSinglePacked(observedObject.transform.rotation.eulerAngles.z);
-                            }
-                            else
-                            {
-                                writer.WriteBool(false);
-                            }
-
-                            if (NetworkConfig.EnableNetworkVariable)
-                            {
-                                observedObject.WriteNetworkVariableData(buffer, ownerClientId);
-                            }
+                            m_ObservedObjects[i].SerializeSceneObject(writer, ownerClientId);
```

Comment:
In the old code, you're allocating the buffer, then  a writer, then use both all the way through.

In the new code, you pass this writer in, and then, rather than using this aforementioned allocated buffer, we call writer.GetStream() to locally set up a buffer.  I'm trying to convince myself this is ok :).  Can you explain a bit?  


 call and allocating a second buffer inside of SerializeSceneObject().  Maybe pass in the buffer too?

Reply:
So, as I understand this:
NetworkWriter.cs
```
        /// <summary>
        /// Changes the underlying stream the writer is writing to
        /// </summary>
        /// <param name="stream">The stream to write to</param>
        public void SetStream(Stream stream)
        {
            m_Sink = stream;
            m_NetworkSink = stream as NetworkBuffer;
        }

        internal Stream GetStream()
        {
            return m_Sink;
        }
```
The writer.GetStream() method just returns the stream (m_Sink) that was assigned to the writer itself.
So, I am confused as to why this would create a completely separate stream than the one already assigned to the NetworkWriter being passed into SerializeSceneObject?

Reply:
Ok, I was hoping it was just the same buffer.  Sorry, the last sentence was a leftover from when I was drafting the comment and probably seemed confusing.